src/Documenter.jl

```diff
@@ -361,15 +351,32 @@ function deploydocs(;
         devbranch = latest
         @info("setting `devbranch` to `$(devbranch)`.")
     end
+    # deprecation/removal of `julia` and `osname` kwargs
+    if julia !== nothing
+        Base.depwarn("the `julia` keyword argument to `Documenter.deploydocs` is " *
+            "removed. Use Travis Build Stages for determining from where to deploy instead. " *
+            "See the section about Hosting in the Documenter manual for more details.", :deploydocs)
+        julia = "nightly"
+    end
+    if osname !== nothing
+        Base.depwarn("the `osname` keyword argument to `Documenter.deploydocs` is " *
+            "removed. Use Travis Build Stages for determining from where to deploy instead. " *
+            "See the section about Hosting in the Documenter manual for more details.", :deploydocs)
+        osname = "linux"
+    end
+    travis_julia = get(ENV, "TRAVIS_JULIA_VERSION", "")
+    travis_osname = get(ENV, "TRAVIS_OS_NAME","")
+    if !(julia == travis_julia && osname == travis_osname)
+        @info("skipping docs deployment.")
+    end
```

Comment:
Should we maybe go all-in and not deploy at all if `julia` and/or `osname` is defined? Since `julia` is mandatory at the moment, it should catch everyone who go to 0.20.0 without updating `make.jl` (may take a while before people notice though). So this should make sure that people don't get weird deployments when they accidentally start using 0.20.